File: sort/Leetcode/java/src/main/java/leetcode/IntersectionTwoArrays.java
package leetcode;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class IntersectionTwoArrays {
    public static void main(String[] args) {
        int[] nums1 = {1,2,2,2,1,1,3};
        int[] nums2 = {2,2,3};

        int[] answer = intersect(nums1, nums2);
        for (int i = 0; i < answer.length; i++) {
            System.out.println(answer[i]);
        }

    }

    public static int[] intersect(int[] nums1, int[] nums2) {
        ArrayList<Integer> answerList = new ArrayList<>();
        Map<Integer, Integer> map = new HashMap<>();

        //Create the Hashmap
        for (int i = 0; i < nums1.length; i++) {
            map.put(nums1[i], map.getOrDefault(nums1[i], 0)+ 1);
        }

        //Check the other Hashmap
        int currentKey;

        for (int i = 0; i < nums2.length; i++) {
            currentKey = nums2[i];
            if (map.containsKey(currentKey)) {
                int currentCount = map.get(currentKey);
                if (currentCount >= 1) {
                    answerList.add(currentKey);
                    currentCount = currentCount - 1;

                    map.replace(currentKey, currentCount);
                }
            }
        }

        int[] answer = new int[answerList.size()];
        for (int i = 0; i < answerList.size(); i++) {
            answer[i] = answerList.get(i);
        }

        return answer;
    }

}

File: sort/Leetcode/javascript/leetcode/intersectionTwoArrays.js
//const nums1 = [1,2,2,1]; 
//const nums2 = [2,2];
let nums1 = [1,2,3,1,2,3,4]; 
let nums2 = [2,3,4];

intersect(nums1, nums2);




function intersect(nums1, nums2) {
    var maxNums = [];
    const maxSize = nums2.length;

    for (let i = 0; i < nums1.length; i++) {
        let compareN1 = [];
        let compareN2 = [];

        if(nums2[0] == nums1[i]) {

            //console.log("Check " + nums2[0] + " " + nums1[i]);
            compareN1 = 

            findMaxSameArray(compareN1, compareN2)
        
        }
    }  
    
};

function findMaxSameArray(compareN1, compareN2) {


}

function compareNums(n1, n2) {
    for (let i = 0; i < n1.length; i++) {
        if(n1[i] !== n2[i]){
            return false;
        }
    }
    return true;
}


//const n1 = [2,2]; 
//const n2 = [2,2];
//const sameNum = compareNums(n1, n2);
//console.log(sameNum);